app/org/sagebionetworks/bridge/services/ConsentService.java

```diff
@@ -268,19 +261,32 @@ public void consentToResearch(Study study, SubpopulationGuid subpopGuid, StudyPa
             withdrawSignatures(account, subpopGuid, withdrewOn);
         }
         accountDao.updateAccount(account, false);
-        
-        String externalId = optionsService.getOptions(account.getHealthCode()).getString(EXTERNAL_IDENTIFIER);
-        
-        if (account.getEmail() != null) {
-            MimeTypeEmailProvider consentEmail = new WithdrawConsentEmailProvider(study, externalId, account, withdrawal,
-                    withdrewOn);
-            sendMailService.sendEmail(consentEmail);
-        }
-        
+
+        sendWithdrawEmail(study, participant.getExternalId(), account, withdrawal, withdrewOn);
+
         // But we don't need to query, we know these are all withdraw.
         return getConsentStatuses(context);
     }
-    
+
+    // Helper method, which abstracts away logic for sending withdraw notification email.
+    private void sendWithdrawEmail(Study study, String externalId, Account account, Withdrawal withdrawal,
+            long withdrewOn) {
+        if (account.getEmail() == null) {
+            // Withdraw email provider currently doesn't support non-email accounts. Skip.
+            return;
+        }
+
+        Boolean consentNotificationEmailVerified = study.isConsentNotificationEmailVerified();
+        if (consentNotificationEmailVerified != null && !consentNotificationEmailVerified) {
```

Comment:
if (consentNotificationEmailVerified != Boolean.TRUE)

Reply:
The equivalent logic would actually be if (consentNotificationEmailVerified == Boolean.FALSE). I've updated this.